### PR TITLE
PARQUET-1496: Update Scala to 2.12

### PR DESCRIPTION
**Pull request by Fokko**
Make sure you have checked _all_ steps below. Updated the tests a bit as well. There were some conflicts between both `test.thrift` files. I've consolidated them to avoid issues.

Updating to Scala 2.12 is required to support Java 11.

### Jira

- [ ] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftParquetReader.java [hunk 1]
```diff
@@ -1,14 +1,14 @@
-/*
```
Comment (nandorKollar) [inline parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftParquetReader.java]: This file has only style changes, would you mind reverting it?

Comment (Fokko) [reply parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftParquetReader.java]: Valid point, done!

Comment (Fokko) [reply parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftParquetReader.java]: I've reverted a couple more unrelated changes.

**Comment by Fokko** [comment]
Does someone have some spare time to review this? Would be great to move Java 11 support forward.

**Comment by Fokko** [comment]
@nandorKollar Are you fine with the current PR as is? 👍

**Comment by nandorKollar** [comment]
Sure, thanks @Fokko!

**Pull request merged**